ztest/ztest.go

```diff
@@ -523,7 +524,7 @@ func runzq(path, zedProgram, input string, outputFlags []string, inputFlags []st
 	if err != nil {
 		return "", err
 	}
-	q, err := runtime.CompileQuery(context.Background(), zctx, compiler.NewCompiler(), ast, []zio.Reader{zrc})
+	q, err := runtime.CompileQuery(context.Background(), zctx, compiler.NewCompiler(storage.NewLocalEngine()), ast, []zio.Reader{zrc})
```

Comment:
Nit: This path doesn't use the engine.
```suggestion
	q, err := runtime.CompileQuery(context.Background(), zctx, compiler.NewCompiler(nil), ast, []zio.Reader{zrc})
```